src/Compilers/Core/Portable/SourceGeneration/SyntaxStore.cs

```diff
@@ -51,6 +53,8 @@ public IStateTable GetSyntaxInputTable(SyntaxInputNode syntaxInputNode, NodeStat
                 // when we don't have a value for this node, we update all the syntax inputs at once
                 if (!_tableBuilder.Contains(syntaxInputNode))
                 {
+                    _causedUpdate = syntaxInputNode;
```

Comment:
For generators that don't consume syntax at all, they'll still be updated or do we need other methods updated too? A generator that consumes nothing but is still stupidly expensive will get flagged? This is a question from the perspective of me not quite knowing the implementation here of this type or what it's for.

Reply:
Yes, this offset calculation only affects generators that touch syntax. Basically first generator to hit a syntax input updates all the others. That means the wall clock time calculated in `GeneratorDriver` here https://github.com/dotnet/roslyn/pull/60267/files#diff-d19793306af7a0ed1d42db723b6eba037523b35a290799c83402e854d7e5917dR261 incorrectly includes syntax timings for other generators. 

Regular generators that don't use syntax will just have an offset adjustment of 0. The 'victim' that ran all the updates gets a negative offset to remove the extra time. Everything else gets the time spend calculating its nodes to ensure its accurate.